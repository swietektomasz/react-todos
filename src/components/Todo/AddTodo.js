import React, { Component } from 'react'

export default class AddTodo extends Component {
  state = {
    text: '',
  }

  render() {
    return (
      <div>
        <input placeholder="add todo..." />
        <button onClick={this.props.handleAdding}>add</button>
      </div>
    )
  }
}
