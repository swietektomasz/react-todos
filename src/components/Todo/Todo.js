import React, { Component } from 'react'

export default class Todo extends Component {
  _handleCheckboxChange = () => {
    this.props.handleChecking(this.props.id)
  }

  render() {
    return (
      <div>
        <label>{this.props.title}</label>
        <input
          type="checkbox"
          onChange={this._handleCheckboxChange}
          defaultChecked={this.props.done}
        />
      </div>
    )
  }
}
