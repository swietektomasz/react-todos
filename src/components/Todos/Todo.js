import React from 'react'
import PropTypes from 'prop-types'

export const Todo = ({ handleChecking, title, done, id }) => (
  <div>
    <label>{title}</label>
    <input type="checkbox" onChange={() => handleChecking(id)} defaultChecked={done} />
  </div>
)

export default Todo

Todo.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  done: PropTypes.bool,
  handleChecking: PropTypes.func,
}
