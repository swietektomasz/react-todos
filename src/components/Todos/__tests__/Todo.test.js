import React from 'react'
import renderer from 'react-test-renderer'

import Todo from '../Todo'

describe('<Todo />', () => {
  test('render', () => {
    const defaultProps = {}
    const wrapper = renderer.create(<Todo {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('on checkbox click calls handleChecking from props', () => {
    const defaultProps = {
      id: 1,
      title: 'test',
      done: false,
      handleChecking: jest.fn(),
    }
    const wrapper = renderer.create(<Todo {...defaultProps} />)
    const root = wrapper.root

    expect(root.findByType('input')).toBeDefined()
    expect(root.findAllByType('label')).toHaveLength(1)

    root.findByType('input').props.onChange()
    expect(defaultProps.handleChecking).toBeCalledWith(defaultProps.id)
  })
})
