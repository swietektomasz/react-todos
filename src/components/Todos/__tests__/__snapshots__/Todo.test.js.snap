exports[`<Todo /> render 1`] = `
<div>
  <label />
  <input
    defaultChecked={undefined}
    onChange={[Function]}
    type="checkbox" />
</div>
`;
